File: api/Enkl.Api/Controllers/AiAssistantController.cs
using Enkl.Api.Auth;
using Enkl.Api.Dtos;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.RateLimiting;

namespace Enkl.Api.Controllers;

/// <summary>
/// v4 Phase 1 AI Assistant — [Authorize(Policy = "ProjectMember")] since it acts on one project's
/// tasks on the caller's behalf; no OrgAdmin gate (any project member can use it, same as the board
/// itself). See AiAssistantService's own doc comment for the tool-loop shape and security model.
/// [EnableRateLimiting("ai-assistant")] (Program.cs) caps per-caller spend, since each request can
/// fan out to several Claude API calls internally.
/// </summary>
[ApiController]
[Authorize(Policy = "ProjectMember")]
[EnableRateLimiting("ai-assistant")]
[Route("api/projects/{projectId:guid}/ai-assistant")]
public class AiAssistantController : ControllerBase
{
    private readonly AiAssistantService _assistant;

    public AiAssistantController(AiAssistantService assistant)
    {
        _assistant = assistant;
    }

    [HttpPost("chat")]
    public async Task<IActionResult> Chat(Guid projectId, AiAssistantChatRequest request)
    {
        try
        {
            var result = await _assistant.ChatAsync(projectId, request, User.UserId(), User.IsOrgAdmin());
            return result is null ? NotFound() : Ok(result);
        }
        catch (AiAssistantNotEntitledException)
        {
            return Forbid();
        }
    }

    /// <summary>Lets the frontend decide whether to show the AI Assistant bubble at all, without
    /// exposing a full chat round-trip just to find out. Polled periodically (features/ai-assistant.js)
    /// rather than fetched once, since a Vendor Portal admin can revoke entitlement mid-session - the
    /// chat endpoint above independently re-checks on every call regardless of what this returns, so
    /// this is a UI convenience only, never the actual enforcement point.</summary>
    [HttpGet("availability")]
    public async Task<IActionResult> Availability(Guid projectId)
    {
        var enabled = await _assistant.IsProjectOrgEntitledAsync(projectId, "ai_assistant");
        return enabled is null ? NotFound() : Ok(new { enabled = enabled.Value });
    }
}

File: php-api/src/Controllers/AiAssistantController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\AiAssistantNotEntitledException;
use Enkl\Api\Services\AiAssistantService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/** Ported from Controllers/AiAssistantController.cs. ProjectMemberMiddleware is applied on the route
 * group in routes.php (same shape as TasksController), not here. */
final class AiAssistantController extends BaseController
{
    private function service(): AiAssistantService
    {
        return new AiAssistantService(Database::connection());
    }

    public function chat(Request $request, Response $response, array $args): Response
    {
        try {
            $result = $this->service()->chat($args['projectId'], $this->body($request), $this->callerUserId($request), $this->callerIsOrgAdmin($request));
            return $result === null ? $this->notFound($response) : $this->json($response, $result);
        } catch (AiAssistantNotEntitledException) {
            return $this->json($response, ['message' => 'AI Assistant is not available for your organisation.'], 403);
        }
    }

    /** Lets the frontend decide whether to show the AI Assistant bubble at all, without a full chat
     * round-trip. Polled periodically (features/ai-assistant.js) - chat() above independently
     * re-checks on every call regardless of what this returns, so this is a UI convenience only. */
    public function availability(Request $request, Response $response, array $args): Response
    {
        $enabled = $this->service()->isProjectOrgEntitled($args['projectId'], 'ai_assistant');
        return $enabled === null ? $this->notFound($response) : $this->json($response, ['enabled' => $enabled]);
    }
}
